### DISCUSSION TITLE
where to set setxkbmap?

### DISCUSSION THREAD

**Discussion by bobblkabb** [leftwm/leftwm#1146] [Q&A] [answered]
Hi, 

I'm running leftwm on void linux and I wonder where and how I have to execute setxkbmap to change the keyboard layout?
In config.rom or fish shell or..?

Or is there any leftwm function to change the keyboard layout?

BR,
Michael

**Comment by blackdragon2447** [comment]
Leftwm will on starting up try to execute `$XDG_CONFIG_HOME/leftwm/up` (usually `~/.config/leftwm/up`) if it exists. My suggestion would be to put the `setxkbmap` in there. (If it does not exist, just create an empty bash script).

**Comment by bobblkabb** [comment]
I put a file called 'up' into the leftwm folder:

```
setxkbmap -layout de
polybar --config=~/.config/polybar/config.ini
```

Doesn't work. What did i do wrong?

**Reply by blackdragon2447** [reply]
Is the file executable?

**Reply by bobblkabb** [reply]
Yes, it is. 
Is it maybe because of a different init system. Void linux operates with runit..

**Reply by blackdragon2447** [reply, marked as answer]
Init doesn't have anything to do with this, calling the up script is on leftwm, not init.
Try adding `#!/usr/bin/env bash` to the top of the file. Also does the file run in you run it manually?

**Comment by bobblkabb** [comment]
Now it works. Thank you very much for the quick response!
